pkg/sql/parser/sql.y

```diff
@@ -4146,16 +4150,52 @@ create_tenant_stmt:
   {
     $$.val = &tree.CreateTenant{Name: tree.Name($3)}
   }
-| CREATE TENANT name FROM REPLICATION OF name ON string_or_placeholder
+| CREATE TENANT name FROM REPLICATION OF name ON string_or_placeholder opt_with_tenant_replication_options
   {
     $$.val = &tree.CreateTenantFromReplication{
       Name: tree.Name($3),
       ReplicationSourceTenantName: tree.Name($7),
       ReplicationSourceAddress: $9.expr(),
+      Options: *$10.tenantReplicationOptions(),
     }
   }
 | CREATE TENANT error // SHOW HELP: CREATE TENANT
 
+// Optional tenant replication options.
+opt_with_tenant_replication_options:
+  WITH tenant_replication_options_list
+  {
+    $$.val = $2.tenantReplicationOptions()
+  }
+| WITH OPTIONS '(' tenant_replication_options_list ')'
```

Comment:
so we can also do `CREATE TENANT .. FROM REPLICATION OF .. ON .. WITH OPTIONS (RETENTION = 36h);` right? can you add a test for that syntax? or maybe we can wait for the data driven tests for that?

btw is that standard to support both `WITH X = y` and `WITH OPTIONS (X = y)`?

Reply:
Added a parser test.

The `WITH OPTIONS (X =y)` is also supported by BACKUP and RESTORE so I supported it here for consistency.